src/components/Molecules/Core/Carousel/Carousel.tsx

```
@@ -0,0 +1,116 @@
+import { useRef, type FC } from "react";
+
+import type { Swiper as SwiperType } from "swiper";
+import { Navigation, Pagination } from "swiper/modules";
+import { Swiper, SwiperSlide } from "swiper/react";
+
+import clsx from "clsx";
+import "swiper/css";
+import "swiper/css/navigation";
+import "swiper/css/pagination";
+
+import styles from "./Carousel.module.scss";
+
+import { ArrowButton } from "@components/ArrowButton/ArrowButton";
+import { ToolCard } from "@components/ToolCard/ToolCard";
+import { CardUnit } from "@components/CardUnit/CardUnit";
+import type {
+  CarouselProps,
+  DynamicCarouselItemComponentProps,
+} from "./Carousel.interface";
+
+// List of components that we allow the Carousel to render
+type ComponentName = "ToolCard" | "CardUnit";
+export const COMPONENT_MAP: Record<ComponentName, FC<any>> = {
+  ToolCard,
+  CardUnit,
+};
+
+export const DynamicCarouselItemComponent: FC<
+  DynamicCarouselItemComponentProps
+> = ({ blok }) => {
+  const component: string = blok.component;
+
+  // If we are trying to render a blok we don't have a corresponding component for, raise error
+  if (!Object.keys(COMPONENT_MAP).includes(component)) {
+    throw new Error(
+      `DynamicComponent error: Component "${component}" not found in COMPONENT_MAP.`,
+    );
+  }
```

I'm wondering whether we should not actually throw an error if a component isn't supported, but just fail silently and log the error, as this breaks the preview as you have seen. Don't worry about it now but if we raise a ticket on the other `DynamicComponent` to change this behaviour we should change it here too.